docs/language/contracts.md

```diff
@@ -225,20 +225,43 @@ Contracts have the implicit field `let account: Account`,
 which is the account in which the contract is deployed too.
 This gives the contract the ability to e.g. read and write to the account's storage.
 
-## Deploying and Updating Contracts
+## Deploying, Updating, and Removing Contracts
+
+### Deploying, Updating, and Removing Contracts
 
 In order for a contract to be used in Cadence, it needs to be deployed to an account.
-A contract can be deployed to an account using the `setCode` function of the `AuthAccount` type:
+The contracts of an account can be accessed through the `contracts` object.
+
+#### Deploying a New Contract
+
+A new contract can be deployed to an account using the `add` function:
 
-- `fun AuthAccount.setCode(_ code: [UInt8], ... contractInitializerArguments)`
+  ```cadence
+  fun add(
+      name: String,
+      code: [UInt8],
+      ... contractInitializerArguments
+  ): DeployedContract
+  ```
+
+  Adds the given contract to the account.
+
+  The `code` parameter is the UTF-8 encoded representation of the source code.
+  The code must contain exactly one contract or contract interface,
+  which must have the same name as the `name` parameter.
 
-  The `code` parameter is the byte representation of the source code.
   All additional arguments that are given are passed further to the initializer
   of the contract that is being deployed.
 
+  Fails if a contract/contract interface with the given name already exists in the account,
+  if the given code does not declare exactly one contract or contract interface,
+  or if the given name does not match the name of the contract/contract interface declaration in the code.
+
+  Returns the deployed contract.
```

Comment:
Note the return type `DeployedContract`, that's what I meant with "deployed contract" – that's the code of the deployed contract, not the contract singleton object (the later is not possible, because the checker can't know what type is being deployed statically)

Reply:
I have to clarify that, thanks for the question!

Reply:
Maybe `DeployedCode` or a totally different name is less confusing? 🤔

Reply:
Tried to explain/clarify it in 130ed31 by adding a subsection and linking to it everywhere.